File: leetcode/power-of-two.c
// Given an integer, write a function to determine if it is a power of two.

#include <stdio.h>

int
ispow2(int x)
{
	return x > 0 && (x & (x - 1)) == 0;
}

int
main(void)
{
	int i;

	printf("%d\n", ispow2(1));
	printf("%d\n", ispow2(16));
	printf("%d\n", ispow2(218));
	printf("\n");
	for (i = 0; i <= 1 << 27; i++) {
		if (ispow2(i))
			printf("%x\n", i);
	}
	return 0;
}

File: leetcode/power-of-two.rs
/*

Given an integer, write a function to determine if it is a power of two.

*/

fn main() {
    for i in 1..8193 {
        if is_power_of_two(i) {
            println!("{}", i);
        }
    }
}

fn is_power_of_two(n: usize) -> bool {
    if n == 0 {
        false
    } else {
        n&(n-1) == 0
    }
}
